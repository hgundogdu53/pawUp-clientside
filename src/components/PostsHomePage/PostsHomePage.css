h1.main-title-post {
    margin: 1rem;
}

h4.pet_name {
    font-size: 1.55rem;
    animation-name: moveInLeft;
    animation-duration: 2s;
    animation-timing-function: ease-in;
}

a.email {
    color: #d7efea;
}

a.email:hover {
    color: #c4f39f;
}

.post-main {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}
.bm-menu-wrap {
    position: fixed ;
    height: 100%;
  }
  
  
.card {
  display:inline-block;
    background-color: rgba(93, 136, 168, 0.616);
    transition: 0.3s;
    border-radius: 5px;
    width: 22.5rem;
    margin-top: 0.1rem;
    
}
  
.card:hover {
    box-shadow: 0 8px 16px 0 rgba(255, 255, 255, 0.151);
}
  
  
.container {
    border-top: #c4f39f 0.5px solid;
    padding: 0.1rem 1.5rem;
}

  @media only screen and (min-width: 600px) {
      .post-main {
          display: grid;
          grid-gap: 0.2rem 0.25rem;
          grid-template-columns: auto auto;
          justify-content: space-evenly;
          overflow: hidden;
          grid-template-areas: 'myArea myArea';
      }
  }

  @media only screen and (min-width: 1200px) {
    .post-main {
        display: grid;
        grid-gap: 0.5rem 0.25rem;
        grid-template-columns: auto auto auto;
        justify-content: space-evenly;
        overflow: hidden;
        grid-template-areas: 'myArea myArea';
    }
}

  @keyframes moveInLeft {
    0% {
      opacity: 0;
      transform: translateX(-100px);
    }

    80% {
      transform: translateX(10px);
    }

    100% {
      opacity: 1;
      transform: translate(0);
    }
  }