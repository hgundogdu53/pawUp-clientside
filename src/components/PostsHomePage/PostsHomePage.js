import React from 'react';
import moment from 'moment';
import './PostsHomePage.css';

function PostsHomePage(props) {
    return (
        <main>
            <div role="banner">
                <h1 className='main-title-post'>PawUp Community Pets</h1>
            </div>
            <section className='post-main'>
                {props.posts.map(post => {
                    return (
                        <section key={post.id} className='card' >
                            <h4 className='pet_name'>{post.pet_name}</h4>
                            <section className='container'>
                                <p>Owner Email: <a href={`mailto:${post.email}?subject=Interested in ${post.pet_name}!`} className='email'>{post.email}</a></p>
                                <p>Birthdate: {moment(post.birthdate).format("MM-DD-YYYY")}</p>
                                <p>Location: {post.location}</p>
                                <p>Type of Pet: {post.type_of_pet}</p>
                                <p>Hobbies: {post.hobbies}</p>
                            </section>
                        </section>
                    )
                })}
            </section>
        </main>
    )
}

export default PostsHomePage;