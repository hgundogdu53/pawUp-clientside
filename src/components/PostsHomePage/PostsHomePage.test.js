import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PostsHomePage from './PostsHomePage';

describe(`PostsHomePage component`, () => {
    it.skip('renders without crashing', () => {
        const post = [
            {
                pet_name: 'muffin',
                type_of_pet: 'cat',
                birthdate: '2018-06-18',
                hobbies: 'Small dog that loves to run around. Will fit well in smaller homes as well as larger homes. Will need atleast an hour of exercise to keep out of trouble.',
                owner_id: 1
            },
            {
                pet_name: 'muffin2',
                type_of_pet: 'cat2',
                birthdate: '2019-06-18',
                hobbies: 'funny dog that loves to run around. Will fit well in smaller homes as well as larger homes. Will need atleast an hour of exercise to keep out of trouble.',
                owner_id: 2
            },

        ]
        const wrapper = shallow(<PostsHomePage posts={post.id} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});