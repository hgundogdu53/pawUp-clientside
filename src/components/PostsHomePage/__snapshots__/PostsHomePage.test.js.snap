// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsHomePage component renders without crashing 1`] = `
<main>
  <div
    role="banner"
  >
    <h1
      className="main-title-post"
    >
      PawUp Community Pets
    </h1>
  </div>
  <section
    className="post-main"
  >
    <section
      className="card"
    >
      <h4
        className="pet_name"
      >
        muffin
      </h4>
      <section
        className="container"
      >
        <p>
          Owner Email: 
          <a
            className="email"
            href="mailto:hgundogdu53@gmail.com?subject=Interested in muffin!"
          >
            hgundogdu53@gmail.com
          </a>
        </p>
        <p>
          Birthdate: 
          06-18-2018
        </p>
        <p>
          Location: 
        </p>
        <p>
          Type of Pet: 
          cat
        </p>
        <p>
          Hobbies: 
          Small dog that loves to run around. Will fit well in smaller homes as well as larger homes. Will need atleast an hour of exercise to keep out of trouble.
        </p>
      </section>
    </section>
  </section>
</main>
`;
